CMakeLists.txt

```diff
@@ -62,6 +62,10 @@ option(SWIFT_INCLUDE_DOCS
     "Create targets for building docs."
     TRUE)
 
+option(SWIFT_HOST_COMPILER_ONLY
```

Comment:
This name (outside of documentation context) can be interpreted as being related to the cross-compilation (something like "build only the compiler for the host, not for the target").

Could we use a more explicit name like `SWIFT_BUILD_RUNTIME_WITH_HOST_COMPILER`?